src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreWorker.cs

```diff
@@ -604,26 +606,41 @@ private async Task PromoteTaskToActiveAsync(BackgroundRestoreOperation restoreOp
                 {
                     var componentModel = await _componentModel.GetValueAsync(jobCts.Token);
 
-                    using (var logger = componentModel.GetService<RestoreOperationLogger>())
+                    // Setup MEF container.
+                    AggregateCatalog catalog = new AggregateCatalog(
+                        new AssemblyCatalog(System.Reflection.Assembly.GetExecutingAssembly()),
+                        new AssemblyCatalog(typeof(OutputConsoleProvider).Assembly));
+                    CompositionContainer container = new CompositionContainer(catalog);
+                    container.ComposeParts(this);
```

Comment:
Unfortunately I don't have sufficient expertise to understand any potential side effects from this implementation. 

Maybe we can get someone that understand MEF in VS better to look at this.